spec/requests/api/v1/customers/applied_coupons_spec.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Api::V1::Customers::AppliedCouponsController, type: :request do
+  let(:organization) { create(:organization) }
+  let(:customer) { create(:customer, organization:) }
+  let(:coupon) { create(:coupon, organization:) }
+
+  describe 'destroy' do
+    let(:applied_coupon) { create(:applied_coupon, customer:, coupon:) }
+
+    before { applied_coupon }
+
+    it 'terminates the applied coupon' do
+      expect do
+        delete_with_token(organization, "/api/v1/customers/#{customer.external_id}/applied_coupons/#{coupon.code}")
+      end.to change { applied_coupon.reload.status }.from('active').to('terminated')
+    end
+
+    it 'returns the applied_coupon' do
+      delete_with_token(organization, "/api/v1/customers/#{customer.external_id}/applied_coupons/#{coupon.code}")
+
+      expect(response).to have_http_status(:success)
+      expect(json[:applied_coupon][:lago_id]).to eq(applied_coupon.id)
+    end
+
+    context 'when customer does not exist' do
+      it 'returns not_found error' do
+        delete_with_token(organization, "/api/v1/customers/unknown/applied_coupons/#{coupon.code}")
+
+        expect(response).to have_http_status(:not_found)
+      end
+    end
+
+    context 'when coupon does not exist' do
+      it 'returns not_found error' do
+        delete_with_token(organization, "/api/v1/customers/#{customer.external_id}/applied_coupons/unknown")
+
+        expect(response).to have_http_status(:not_found)
+      end
+    end
+
+    context 'when coupon is not applied to customer' do
+      it 'returns not_found error' do
+        other_coupon = create(:coupon, organization:)
+        delete_with_token(organization, "/api/v1/customers/#{customer.external_id}/applied_coupons/#{other_coupon.code}")
```

Comment:
Layout/LineLength: Line is too long. [121/120] (https://rubystyle.guide#max-line-length)